service/src/main/java/tds/config/services/impl/TimeLimitConfigurationServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package tds.config.services.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import tds.config.TimeLimitConfiguration;
+import tds.config.repositories.TimeLimitConfigurationRepository;
+import tds.config.services.TimeLimitConfigurationService;
+
+import java.util.Optional;
+
+@Service
+public class TimeLimitConfigurationServiceImpl implements TimeLimitConfigurationService {
+    private final TimeLimitConfigurationRepository timeLimitConfigurationRepository;
+
+    @Autowired
+    public TimeLimitConfigurationServiceImpl(TimeLimitConfigurationRepository timeLimitConfigurationRepository) {
+        this.timeLimitConfigurationRepository = timeLimitConfigurationRepository;
+    }
+
+    @Override
+    public Optional<TimeLimitConfiguration> findTimeLimitConfiguration(String clientName) {
+        return timeLimitConfigurationRepository.findTimeLimitConfiguration(clientName);
+    }
+
+    @Override
+    public Optional<TimeLimitConfiguration> findTimeLimitConfiguration(String clientName, String assessmentId) {
+        return timeLimitConfigurationRepository.findTimeLimitConfiguration(clientName, assessmentId)
+                .map(Optional::of)
```

Comment:
Yep - I expected it to compile also.  I tried changing the code to this:
```
    @Override
    public Optional<TimeLimitConfiguration> findTimeLimitConfiguration(String clientName, String assessmentId) {
        return timeLimitConfigurationRepository.findTimeLimitConfiguration(clientName, assessmentId)
                .orElseGet(() -> timeLimitConfigurationRepository.findTimeLimitConfiguration(clientName));
    }
```
Then I ran `mvn clean install -f service/pom.xml`, and here's the error that comes back:
```
[ERROR] /Users/jjohnson/dev/sbac/git/repositories/TDS_ConfigService/service/src/main/java/tds/config/services/impl/TimeLimitConfigurationServiceImpl.java:[29,93] incompatible types: bad return type in lambda expression
[ERROR] java.util.Optional<tds.config.TimeLimitConfiguration> cannot be converted to tds.config.TimeLimitConfiguration
```

So I believe the `orElseGet()` is expecting to return something not optional.  I tried setting the return type of the lambda to `Supplier<Optional<TimeLimitConfiguration>>`, but I couldn't get that to work (granted I only tried for @ a minute or so).

Anyway, I think this reads a bit cleaner:
```
@Override
    public Optional<TimeLimitConfiguration> findTimeLimitConfiguration(String clientName, String assessmentId) {
        Optional<TimeLimitConfiguration> maybeTimeLimitConfig =
                timeLimitConfigurationRepository.findTimeLimitConfiguration(clientName, assessmentId);

        if (maybeTimeLimitConfig.isPresent()) {
            return maybeTimeLimitConfig;
        }

        // RULE:  If time limit configuration cannot be found for clientName + assessmentId, get time limit configuration
        // for clientName.
        return timeLimitConfigurationRepository.findTimeLimitConfiguration(clientName);
    }
```